### PR TITLE
Add TypeScript type definitions

### PR DESCRIPTION
**Pull request by mcmire**
In lieu of converting this whole repo to TypeScript, provide types so that we can type the proxy objects correctly in the `core` NetworkController.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+declare module '@metamask/swappable-obj-proxy' {
+  export type SwappableProxy<T = unknown> = T & {
+    setTarget: (newTarget: T) => void,
+  };
+
+  export function createEventEmitterProxy<T = unknown>(
```
Comment (Gudahtt) [inline index.d.ts:6]: Nit: Should we require that this type extends `EventEmitter`? It does assume the target has EventEmitter methods, and it invokes them.

**Review by Gudahtt — APPROVED** [review]
LGTM! Left one suggestion to improve the specificity of the type

File: index.d.ts (hunk 1 above)
Comment (mcmire) [reply index.d.ts]: Oh, yes, I suppose you're right. I also found that the type for `initialTarget` was wrong. Fixed in [9835937](https://github.com/MetaMask/swappable-obj-proxy/pull/13/commits/9835937634e4d9f76e28e0768596430959578233).

File: package-lock.json [hunk 2]
```diff
@@ -1,16 +1,285 @@
 {
-  "name": "@metamask/swappable-obj-proxy",
+  "name": "swappable-obj-proxy",
   "version": "1.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (Gudahtt) [inline package-lock.json:4]: Hmm.... looks like this was generated with an older version of npm?

Comment (mcmire) [reply package-lock.json:4]: This got reverted by mistake. But this also isn't right on `master` either. I've submitted a new PR to fix it ahead of this PR: https://github.com/MetaMask/swappable-obj-proxy/pull/20

Comment (mcmire) [reply package-lock.json]: Okay, these changes should have never been there in the first place on this branch. They've now been removed.

**Comment by mcmire** [comment]
@Gudahtt Ready for another review!

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**